### PR TITLE
add matching Microsoft.Extensions dependency requirement per target net framework

### PR DESCRIPTION
**Pull request by Havunen**
Fixes https://github.com/serilog/serilog-extensions-logging/issues/229

### CODE REVIEW

File: src/Serilog.Extensions.Logging/Serilog.Extensions.Logging.csproj [hunk 1]
```diff
@@ -30,9 +30,23 @@
...
+  <ItemGroup Condition="'$(targetframework)' == 'net7.0'">
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="7.0.0-*" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(targetframework)' == 'net6.0'">
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0-*" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(targetframework)' == 'netstandard2.1' or '$(targetframework)' == 'netstandard2.0' or '$(targetframework)' == 'net462'">
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="8.0.0-*" />
```
Comment (Havunen) [inline src/Serilog.Extensions.Logging/Serilog.Extensions.Logging.csproj:49]: I'm not exactly sure what should be the version defined here, but at least this does not change the current behavior by using the latest version

**Comment by Numpsy** [comment]
Think this sort of change was already rejected in https://github.com/serilog/serilog-extensions-logging/pull/224

**Comment by nblumhardt** [comment]
Hi! Thanks for raising these. From 7.0.* onwards, n all the packages with versions matching a Microsoft.* package, the version policy is to match major and minor version numbers. If you need to target an earlier Microsoft.* package, picking the matching earlier version of the Serilog package (or next lowest version, for 6.0.* and before) will do what you need.

Closing this as duplicate, but thanks for taking the time to send a PR 👍

**Comment by Havunen** [comment]
@nblumhardt Ok that makes sense, but why is the latest version then targeting old .NET frameworks? Shouldn't it target only the matching version not to confuse users?

**Comment by Havunen** [comment]
Maybe this behavior could be documented in the main readme too?

**Comment by nblumhardt** [comment]
Thanks for your reply. The targeted frameworks match the ones used by _Microsoft.Extensions.Logging_ at the same version (we're using exactly the same dependency and targeting scheme for this package as they do - which is the reason for the current policy). Hope this helps; will check out the README when I have a chance 👍

**Pull request closed without merging**